umbrellapanel-chief/src/main/java/ms/idrea/umbrellapanel/chief/UmbrellaPanelUserDatabase.java

```diff
@@ -6,22 +6,26 @@
 import java.io.Reader;
 import java.io.Writer;
 import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.Collections;
 import java.util.List;
-import java.util.Map;
+
+import com.flowpowered.networking.Message;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
 
 import ms.idrea.umbrellapanel.api.chief.PanelUserDatabase;
 import ms.idrea.umbrellapanel.api.chief.net.NetworkServer;
 import ms.idrea.umbrellapanel.api.core.permissions.PanelUser;
 import ms.idrea.umbrellapanel.net.messages.UpdatePanelUserMessage;
 import ms.idrea.umbrellapanel.net.messages.UpdatePanelUserMessage.Action;
 
-import com.flowpowered.networking.Message;
-
 public class UmbrellaPanelUserDatabase implements PanelUserDatabase {
 
+	private static final JsonParser PARSER = new JsonParser();
 	private NetworkServer networkServer;
-	private Map<Integer, PanelUser> users = new HashMap<>();
+	private final List<PanelUser> users = Collections.synchronizedList(new ArrayList<PanelUser>());
```

Comment:
You should improve the synchronisation.

Reply:
What do you mean exactly? 
Another concurrenc Handling?

The only alternativ should be an CopyOnWriteList.
But in this case is an synchronizedList enough.